### PR TITLE
Deprecate Upload task

### LINKED ISSUE #10796: Fully deprecate the Upload task and related concepts
While Gradle 6.0 deprecates the usage of the `maven` plugin and the usage of the default `Upload` tasks created in a project, there is no general deprecation of the usage of `Upload` tasks.

This should be done to completely retire the legacy publication system.

### PR DESCRIPTION
**Pull request by donat**
Fixes #10796

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -179,3 +179,11 @@ property instead.
 property has been deprecated.
 Use the link:{groovyDslPath}/org.gradle.api.tasks.scala.ScalaCompile.html#org.gradle.api.tasks.scala.ScalaCompile:destinationDirectory[ScalaCompile.destinationDirectory]
 property instead.
+
+[[upload_task_deprecation]]
+=== Deprecated `Upload` task
+
+Gradle used to have two way of publishing artifacts.
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:186]: ways*

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 2]
```diff
@@ -179,3 +179,11 @@ property instead.
...
+Now, the situation has been cleared and all build should use the `maven-publish` plugin.
+The last remaining artifact of the old way of publishing is the `Upload` task that has been deprecated and scheduled for removal in Gradle 8.0.
+Existing clients should migrate to the `maven-publish` plugin.
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:189]: Maybe add a link to the `maven-publish` docs or sample from here?

**Review by rieske — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 1 above)
Comment (donat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Fixed.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 2 above)
Comment (donat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Done.

**Comment by donat** [comment]
@bot-gradle test and merge

**Comment by donat** [comment]
@bot-gradle test and merge

**Comment by donat** [comment]
@bot-gradle test and merge

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

**Pull request merged**